packages/ai-scanoss/src/browser/ai-scanoss-code-scan-action.tsx

```diff
@@ -119,15 +118,12 @@ const ScanOSSIntegration = React.memo((props: {
         }
     }, []);
     const scanOSSClicked = React.useCallback(async () => {
-        let scanResult = scanOSSResult;
-        if (scanResult === 'pending') {
+        if (scanOSSResult === 'pending') {
             return;
-        }
-        if (!scanResult || scanResult.type === 'error') {
-            scanResult = await scanCode();
-        }
-        if (scanResult && scanResult.type === 'match') {
-            const dialog = new ScanOSSDialog(scanResult);
+        } else if (!scanOSSResult || scanOSSResult.type === 'error') {
+            scanCode();
```

Comment:
This is a behavioral change. Now I need to click twice to see the results if they did not exist before. Please restore the previous behavior.

Reply:
Hm... this was deliberate. In general, I think opening a modal as a result of an asynchronous action is not a great experience for the user: if the data isn't available instantly, then the user isn't in control of when the dialog actually appears, meaning they either have to sit and wait, or it can interrupt them if they move on to another task.

Reply:
ScanOSS is very fast (it's basically a hash table lookup), so in the majority of cases the result is there immediately and the user needs to perform a "double click" to open the dialog. The only case this "fixes" is when the user is on Electron and has poor internet reception.

Suggestion: Let's measure the elapsed time until the response is received and if it's below some threshold (e.g. 500ms) then let's open the modal immediately without requiring a further click. This should cover the majority of typical usage scenarios.

Only if it takes longer we don't open the dialog automatically. The user is informed about this as the button also does not switch to a result state and therefore indicates that the result is not there yet.